### ISSUE TITLE
Card - Header & Container should use correct semantic element

### ISSUE THREAD

**Issue by samuelgoff** [grommet/hpe-design-system#2279]
`Card` `Header` & `Container` use incorrect semantic elements, which undermines the most basic accessibility requirements & the ability of assistive technologies to understand/represent/navigate how this content relates to other content on the page.

[Related slack discussion thread](https://grommet.slack.com/archives/C04LMJWQT/p1643385194068500)

### Is this issue related to the design system core library or design system website?
design-system-core  (Grommet components)

### Expected Behavior
- `Card` `Container` should be `div`, [`section`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section#usage_notes), or `article`, depending on what is the best semantic tool for the job), or `article`, depending on what is the best semantic tool for the job
- `Card` `Header` uses a `header`, but no `h1`-`h6` element. `Card` `Header` should contain a semantic `h1`-`h6` element, depending on context.
For example:
- if `Card` `Container` is a `section` (see link above), `Card` `Header` should contain an `h1`
- if `Card` `Container` is a `div` or other semantic element, `h2` might be more appropriate
- If this card is one of many appearing inside a block of content below an `h2`, it would be appropriate for this to be an `h3`, etc.

This relates to the following WCAG 2.1 requirements at the following levels:
- [1.3.1 Info and Relationships - Level A](https://www.w3.org/WAI/WCAG21/quickref/#info-and-relationships)
> Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.
- [2.4.6 Headings and Labels - Level AA](https://www.w3.org/WAI/WCAG21/quickref/#headings-and-labels)
> Headings and labels describe topic or purpose.
- [2.4.10 Section Headings - Level AAA](https://www.w3.org/WAI/WCAG21/quickref/#section-headings)
> Section headings are used to organize the content.
> Note 1: "Heading" is used in its general sense and includes titles and other ways to add a heading to different types of content.

Note that while it may not be "required" to achieve 100% WCAG 2.1/2.2 AAA accessibility, the guidelines themselves speak to making every effort to achieve the highest level possible, wherever it **_is_** possible (i.e. don't just try to meet AA & calling it a day). Of all the many AAA requirements to aim for, IMO, this is a particularly important one, because it can be a game-changer for how easily assistive technologies can make sense of how content relates to each other in a hierarchy, so it becomes much easier to navigate.

### Actual Behavior

- `Card` `Container` currently use a `Box`, which translates into a `div`, regardless of whether each `Card` relates to each other or not.
- `Card` `Header` uses a `header`, but no `h1`-`h6` element

### URL, screen shot, or CodeSandbox exhibiting the issue

Note the `Card` `Header` is a `header` element, and does not contain a `h1`-`h6` of any kind here
![card-header-is-a-header-no-h1-or-h2](https://user-images.githubusercontent.com/2007084/151603444-ce32d4c7-96aa-493a-a639-adf48bf03539.png)

Note the `Card` container element is always a `div`, not a `section`, `article`, or other semantic element.
![card-container-is-a-div](https://user-images.githubusercontent.com/2007084/151603665-d2ef2918-4e84-4ccf-ad6e-29dafbf9fca6.png)

### Steps to Reproduce

1. go to https://v2.grommet.io/card
2. right-click on the `Header` example
3. click `inspect`
4. note the DOM structure & semantic element rendered is a `header`, and contains no `h1`-`h6`
5. walk up the ancestors within the DOM structure & note the `Card` container element is a `div`

### Your Environment
This occurs regardless of browser, OS, etc. however, this is what I'm using:

- Design System version:
- Grommet version: Whatever's deployed to the production [Grommet Card](https://v2.grommet.io/card) example
- Browser Name and version:
Safari Version 15.2 (17612.3.6.1.6)
Chrome Version 97.0.4692.99 (Official Build) (arm64)
- Operating System and version (desktop or mobile):
macOS Monterey Version 12.1 (21C52)

**Comment by taysea** [comment]
Closing, we now recommend using headings within cards: https://design-system.hpe.design/components/card?q=card#card-titles
